xapian-letor/api/featurelist.cc

```diff
@@ -108,23 +108,26 @@ FeatureList::create_feature_vectors(const Xapian::MSet & mset,
     LOGCALL(API, std::vector<FeatureVector>, "FeatureList::create_feature_vectors", mset | letor_query | letor_db);
     if (mset.empty())
 	return vector<FeatureVector>();
+
+    if (!internal->feature.empty()) {
```

Comment:
The sense of this conditional is wrong, surely? We want to throw the exception if `internal->feature` _is_ empty. It looks like the test below should catch this — are you running the tests locally before pushing?

@ojwb when we squash / merge this, we should probably rewrite the commit message to lead with "Raise exception on empty FeatureList", then explain it fixes memory leaks in the commit body.

However this makes me think: shouldn't we actually check this on the `FeatureList` constructor, and then turn this here into an assertion?

Reply:
> It looks like the test below should catch this

CI logs indeed show the new test (and a lot of existing tests) failing due to this.

Checking in the constructor probably makes more sense.  It needs a little care as the destructor won't be called if the constructor throws an exception - probably cleanest to delay the allocation of `internal` until after the check.

Reply:
> It looks like the test below should catch this — are you running the tests locally before pushing?
>
Ah I was so sure that I didn't check it. Fixed in a fixup commit.